staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/etcd.go

```diff
@@ -160,12 +160,17 @@ type StatusREST struct {
 	store *genericregistry.Store
 }
 
-var _ = rest.Updater(&StatusREST{})
+var _ = rest.Patcher(&StatusREST{})
```

Comment:
Can you add this line to the [custom resource `StatusREST`](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go#L115) as well? It currently satisfies the `Patcher` interface implicitly and supports get/update/patch but I think explicitly mentioning it is nicer and safer.

Also, `StatusREST` for [deployments](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/apps/deployment/storage/storage.go#L117), [statefulsets](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/apps/statefulset/storage/storage.go#L98), etc does not implement `Patcher` explicitly as well...but I guess we should add it at least for the custom resource.

Reply:
added